### PR TITLE
Refs:#5068 Bugfix/callback issue

### PR DESCRIPTION
**Pull request by MiguelBarro**
The ParticipantImpl::~ParticipantImpl() must be modified to prevent a discovery callback error. 

As implementation currently works the Participant object pass as argument to the callbacks is destroyed before this callbacks are cutoff. Callbacks are cutoff in the RTPSDomain::removeRTPSParticipant so the Participant object pass as argument should at least be alive till this call returns. The file src/cpp/participant/ParticipantImpl.cpp should be modified by reordering:

			// delete(mp_participant); // former position

			if(this->mp_rtpsParticipant != nullptr)
			{
				RTPSDomain::removeRTPSParticipant(this->mp_rtpsParticipant);
			}

			delete(mp_participant); // new position

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1478/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1444/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1455/)

File: src/cpp/rtps/builtin/discovery/participant/PDPSimpleListener.cpp [hunk 1]
```diff
@@ -149,6 +149,8 @@ void PDPSimpleListener::onNewCacheChangeAdded(RTPSReader* reader, const CacheCha
             {
                 listener->onParticipantDiscovery(this->mp_SPDP->getRTPSParticipant()->getUserRTPSParticipant(), std::move(info));
             }
+
+            return; // change already removed from history
```
Comment (richiware) [inline src/cpp/rtps/builtin/discovery/participant/PDPSimpleListener.cpp]: Then, where is it removed?

Comment (MiguelBarro) [reply src/cpp/rtps/builtin/discovery/participant/PDPSimpleListener.cpp]: Is is removed inside the successful PDPSimple::removeRemoteParticipant call.

**Comment by richiware** [comment]
Failed tests not related.

**Pull request merged**